packages/geo/src/lib/print/shared/print.service.ts

```diff
@@ -744,25 +662,180 @@ export class PrintService {
+   * @returns promise of new canvas
+   */
+  private async drawMap(
+    size: Array<number>,
+    mapCanvas: any
+  ): Promise<HTMLCanvasElement> {
+    return new Promise( async (res) => {
+      const mapResultCanvas = document.createElement('canvas');
+      const mapContextResult = mapResultCanvas.getContext('2d');
+      mapResultCanvas.width = size[0];
+      mapResultCanvas.height = size[1];
+      const opacity = mapCanvas.parentNode.style.opacity || mapCanvas.style.opacity;
+      mapContextResult.globalAlpha = opacity === '' ? 1 : Number(opacity);
+      let matrix;
+      const transform = mapCanvas.style.transform;
+      if (transform) {
+        // Get the transform parameters from the style's transform matrix
+        matrix = transform
+          .match(/^matrix\(([^\(]*)\)$/)[1]
+          .split(',')
+          .map(Number);
+      } else {
+        matrix = [
+          parseFloat(mapCanvas.style.width) / mapCanvas.width,
+          0,
+          0,
+          parseFloat(mapCanvas.style.height) / mapCanvas.height,
+          0,
+          0,
+        ];
+      }
+      // Apply the transform to the export map context
+      CanvasRenderingContext2D.prototype.setTransform.apply(
+        mapContextResult,
+        matrix
+      );
+      const backgroundColor = mapCanvas.parentNode.style.backgroundColor;
+      if (backgroundColor) {
+        mapContextResult.fillStyle = backgroundColor;
+        mapContextResult.fillRect(0, 0, mapCanvas.width, mapCanvas.height);
+      }
+      mapContextResult.drawImage(mapCanvas, 0, 0);
+      mapContextResult.globalAlpha = 1;
+      // reset canvas transform to initial
+      mapContextResult.setTransform(1, 0, 0, 1, 0, 0);
+      res(mapResultCanvas);
+    });
+  }
+
+  /**
+   * this function to draw map scaleLine, attribution and north direction arrow
+   * @param map IgoMap
+   * @param canvas HTMLCanvasElement
+   * @param position legend position
+   * @returns {Promise} Promise object represents HTMLCanvasElement
+   */
+  private async drawMapControls (
+    map: IgoMap,
+    canvas: HTMLCanvasElement,
+    position: PrintLegendPosition
+  ): Promise<HTMLCanvasElement> {
+    return new Promise( async (res) => {
+      const context = canvas.getContext('2d');
+      // Get the scale and attribution
+      // we use cloneNode to modify the nodes to print without modifying it on the page, using deep:true to get children
+      let canvasOverlayHTML;
+      const mapOverlayHTML = map.ol.getOverlayContainerStopEvent().cloneNode(true) as HTMLElement;
+      // to add North Direction to mapOverly
+      await this.setUpNorthDirection(mapOverlayHTML, position);
+      // set up map Attribution designe to print
+      await this.setUpAttribution(mapOverlayHTML);
+
+      // set 'OverlayContainer' size to print size
+      mapOverlayHTML.style.width = canvas.width+'px';
+      mapOverlayHTML.style.height = canvas.height+'px';
+      // we add zindex -1 to not show modification to the user
```

Comment:
J'aime bien ces commentaires:
- // we add zindex -1 to not show modification to the user
- // Append to 'ol-viewport' to load all design make sure html2canvas to render the image correctly

Je propose de faire le ménage des autres commentaires qui ne sont pas essentiels. Par exemple, on retrouve encore des références au 'print-area'. Ceux que je trouve moins essentiels:
- // set up map Attribution designe to print (La méthode le mentionne déjà)